userspace/libsinsp/filter/ast.cpp

```diff
@@ -180,64 +180,62 @@ std::string libsinsp::filter::ast::as_string(ast::expr &e)
 	return sv.as_string();
 }
 
-expr* libsinsp::filter::ast::clone(expr* e)
+std::unique_ptr<expr> libsinsp::filter::ast::clone(expr* e)
 {  
     struct clone_visitor: public expr_visitor
     {   
-        expr* m_last_node;
+        std::unique_ptr<expr> m_last_node;
 
         void visit(and_expr* e) override
         {
-            std::vector<expr*> children;
+            std::vector<std::unique_ptr<expr>> children;
             for (auto &c: e->children)
             {
                 c->accept(this);
-                children.push_back(m_last_node);
+                children.push_back(std::move(m_last_node));
             }
-            m_last_node = new and_expr(children);
+            m_last_node = and_expr::create(children);
         }
 
         void visit(or_expr* e) override
         {
-            std::vector<expr*> children;
+            std::vector<std::unique_ptr<expr>> children;
             for (auto &c: e->children)
             {
                 c->accept(this);
-                children.push_back(m_last_node);
+                children.push_back(std::move(m_last_node));
             }
-            m_last_node = new or_expr(children);
+            m_last_node = or_expr::create(children);
         }
 
         void visit(not_expr* e) override
         {
             e->child->accept(this);
-            m_last_node = new not_expr(m_last_node);
+            m_last_node = not_expr::create(std::move(m_last_node));
         }
 
         void visit(binary_check_expr* e) override
         {
             e->value->accept(this);
-            m_last_node = new binary_check_expr(
-                e->field, e->arg, e->op, m_last_node);
+            m_last_node = binary_check_expr::create(e->field, e->arg, e->op, std::move(m_last_node));
         }
 
         void visit(unary_check_expr* e) override
         {
-            m_last_node = new unary_check_expr(e->field, e->arg, e->op);
+            m_last_node = unary_check_expr::create(e->field, e->arg, e->op);
         }
 
         void visit(value_expr* e) override
         {
-            m_last_node = new value_expr(e->value);
+            m_last_node = value_expr::create(e->value);
         }
 
         void visit(list_expr* e) override
         {
-            m_last_node = new list_expr(e->values);
+            m_last_node = list_expr::create(e->values);
         }
     } visitor;
 
-    visitor.m_last_node = NULL;
```

Comment:
note for the reviewer: this looked redundant, especially because a `unique_ptr` starts null/without any object, but please double check